packages/roosterjs-editor-plugins/lib/plugins/TableResize/TableResize.ts

```diff
@@ -515,159 +524,169 @@ export default class TableResize implements EditorPlugin {
-            if (shouldResizeX || shouldResizeY) {
-                for (let i = 0; i < vtable.cells.length; i++) {
-                    for (let j = 0; j < vtable.cells[i].length; j++) {
-                        const cell = vtable.cells[i][j];
-                        if (cell.td) {
-                            if (shouldResizeX) {
-                                const originalWidth: number = cell.td.style.width
-                                    ? parseFloat(
-                                          cell.td.style.width.substr(
-                                              0,
-                                              cell.td.style.width.length - 2
-                                          )
-                                      )
-                                    : cell.td.getBoundingClientRect().right -
-                                      cell.td.getBoundingClientRect().left;
-                                const newWidth = originalWidth * ratioX;
-                                cell.td.style.boxSizing = 'border-box';
-                                if (newWidth >= MIN_CELL_WIDTH) {
-                                    cell.td.style.wordBreak = 'break-word';
-                                    cell.td.style.width = `${newWidth}px`;
-                                }
+            this.isReadyForResizingTable = true;
+        }
+        const rect = normalizeRect(this.resizingVtable.table.getBoundingClientRect());
+
+        const ratioX =
+            1.0 +
+            (this.isRTL
+                ? (this.currentTableVerticalBorder - mouseX) /
+                  (rect.right - this.currentTableVerticalBorder)
+                : (mouseX - this.currentTableVerticalBorder) /
+                  (this.currentTableVerticalBorder - rect.left));
+        const ratioY =
+            1.0 +
+            (mouseY - this.currentTableHorizontalBorder) /
+                (this.currentTableHorizontalBorder - rect.top);
+
+        const shouldResizeX = Math.abs(ratioX - 1.0) > 1e-3;
+        const shouldResizeY = Math.abs(ratioY - 1.0) > 1e-3;
+        if (shouldResizeX || shouldResizeY) {
+            for (let i = 0; i < this.resizingVtable.cells.length; i++) {
+                for (let j = 0; j < this.resizingVtable.cells[i].length; j++) {
+                    const cell = this.resizingVtable.cells[i][j];
+                    const originalTd = this.cellsBeforeResize[i][j];
+                    if (cell.td && originalTd) {
+                        if (shouldResizeX) {
+                            // the width of some external table is fixed, we need to make it resizeable
+                            this.resizingVtable.table.style.width = null;
+                            const originalWidth: number = parseFloat(
+                                originalTd.style.width.substr(
+                                    0,
+                                    originalTd.style.width.indexOf('px')
```

Comment:
I suggest we never read the value of CSS width/height. That is for browser to read. We can always use either clientWidth/Height or getBoundingClientRect()

Reply:
fixed!